### PR TITLE
Plugin Batch Account Managed Identity

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: plugins/azure/batchAccounts/batchAccountsManagedIdentity.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure/');
+
+module.exports = {
+    title: 'Batch Account Managed Identity',
+    category: 'Batch',
+    domain: 'Compute',
+    severity: 'Medium',
+    description: 'Ensures that Batch account have managed identity enabled.',
```
Comment (alphadev4) [inline plugins/azure/batchAccounts/batchAccountsManagedIdentity.js:9]:
```suggestion
    description: 'Ensures that Batch accounts have managed identity enabled.',
```

File: plugins/azure/batchAccounts/batchAccountsManagedIdentity.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            }
+            if (!batchAccounts.data.length) {
+                helpers.addResult(results, 0, 'No existing Batch accounts found', location);
+                return rcb();
+            }
+
+            for (let batchAccount of batchAccounts.data) { 
+                if (!batchAccount.id) continue;
+        
+                if (batchAccount.identity) {
```
Comment (alphadev4) [inline plugins/azure/batchAccounts/batchAccountsManagedIdentity.js:40]:
```suggestion
                if (batchAccount.identity && batchAccount.identity.type) {
```

File: plugins/azure/batchAccounts/batchAccountsManagedIdentity.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            if (!batchAccounts.data.length) {
+                helpers.addResult(results, 0, 'No existing Batch accounts found', location);
+                return rcb();
+            }
+
+            for (let batchAccount of batchAccounts.data) { 
+                if (!batchAccount.id) continue;
+        
+                if (batchAccount.identity && batchAccount.identity.type) {
+                    helpers.addResult(results, 0, 'Batch account has managed identity enabled', location, batchAccount.id);
```
Comment (alphadev4) [inline plugins/azure/batchAccounts/batchAccountsManagedIdentity.js:41]: can we enable the managed identity as in whole on batch account or pools? from the link it seems like we can enable the identities for the pools in the batch account https://learn.microsoft.com/en-us/azure/batch/managed-identity-pools

Comment (alphadev4) [inline plugins/azure/batchAccounts/batchAccountsManagedIdentity.js:41]: can we enable the managed identity as in whole on batch account or pools? from the link it seems like we can enable the identities for the pools in the batch account https://learn.microsoft.com/en-us/azure/batch/managed-identity-pools

**Review by alphadev4 — CHANGES_REQUESTED** [review]

File: plugins/azure/batchAccounts/batchAccountsManagedIdentity.js (hunk 3 above)
Comment (AkhtarAmir) [reply plugins/azure/batchAccounts/batchAccountsManagedIdentity.js:41]: We can enable it in either of them. Link: https://learn.microsoft.com/en-us/troubleshoot/azure/hpc/batch/use-managed-identities-azure-batch-account-pool

**Review by alphadev4 — APPROVED** [review]

**Pull request merged**